File: services/orchestrator/src/orchestrator/interfaces/api/schemas/report.py
"""
This module defines the `QuestionResult` data class, which represents the result of a single
question in a benchmark test. It includes information about the benchmark, the question, the
expected and actual answers, correctness, score, latency, number of attempts, status, and any
error messages.
"""

from dataclasses import dataclass
from datetime import datetime

from orchestrator.domain.models.metrics import BenchmarkMetrics, Metrics
from pydantic import BaseModel


@dataclass(frozen=True)
class ReportSummary(Metrics):
    total_wall_time_sec: float
    throughput_req_s: float
    benchmarks: list[BenchmarkMetrics]


@dataclass(frozen=True)
class BenchmarkReport(BaseModel):
    generated_at: datetime
    summary: ReportSummary

File: services/frontend/src/types/report.ts

export type QuestionResult = {
    benchmark_id: string
    question_id: string
    question: string
    expected_answer: string
    answer: string | null
    correct: boolean
    score: number
    latency_ms: number | null
    attempts: number
    status: "success" | "failed"
    error: string | null
}




export type LatencyMetrics = {
    p50: number | null
    p95: number | null
    min: number | null
    max: number | null
}

export type Metrics = {
    total_requests: number
    successful_requests: number
    failure_count: number
    accuracy: number
    latency_ms: LatencyMetrics
}

export type BenchmarkMetrics = {
    benchmark_id: string
    metrics: Metrics
}

export type ReportSummary = Metrics & {
    total_wall_time_sec: number
    throughput_req_s: number
    benchmarks: BenchmarkMetrics[]
}

export type BenchmarkReport = {
    generated_at: string
    summary: ReportSummary
}